core/dbt/exceptions.py

```diff
@@ -692,9 +692,11 @@ def missing_materialization(model, adapter_type):
 
 
 def bad_package_spec(repo, spec, error_message):
-    raise InternalException(
-        "Error checking out spec='{}' for repo {}\n{}".format(
-            spec, repo, error_message))
+    msg = "Error checking out spec='{}' for repo {}\n{}".format(spec, repo, error_message)
+    for secret in get_secret_env():
+        msg = str(msg).replace(secret, "*****")
```

Comment:
There's a function for this we can reuse: `functions.py::scrub_secrets` if you call it like this:
```
scrub_secrets(msg, env_secrets())
```

Reply:
In any case, events like this shouldn't be leaking secrets anyway because we are already running this secret scrub across the entire JSON string. And if they're not using json logs, ad-hoc fields like this aren't put into the message.

Reply:
Is there a ticket associated with this leaked secret we're trying to catch?

Reply:
oops my bad. Just catching that this is an exception not an event. this should totally work as a stop gap.